### PR TITLE
[CARBONDATA-3132]Correct the task disrtibution in case of compaction when the actual block nodes and active nodes are different

### PR DESCRIPTION
**Pull request by akashrn5**
### Why This PR?
There is an unequal distribution of tasks during compaction
ex: When the load is done using replication factor 2 and all nodes are active and during compaction one node is down, basically it is not active executor, so the task distribution should take care to distribute the tasks equally among all the active executors instead of giving more tasks to single executor and less to other executor. But sometimes the unequal distribution happens and the compaction becomes sow.

### Solution
Currently we are not getting active executors before the node block mapping and sending the list of active executors as null, which will lead to the above problem sometimes. so get the active executors and send for node block mapping logic which will handle to distribute equally.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 NA
 - [x] Any backward compatibility impacted?
 NA
 - [x] Document update required?
NA

 - [x] Testing done
tested using 3 node and 6 node cluster
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

File: tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java [hunk 1]
```diff
@@ -205,7 +206,7 @@ public void testSummaryOutputAll() {
     expectedOutput = buildLines(
         "## version Details",
         "written_by  Version         ",
-        "TestUtil    1.6.0-SNAPSHOT  ");
+        "TestUtil    "+ CarbonVersionConstants.CARBONDATA_VERSION+"  ");
```
Comment (akashrn5) [inline tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java]: removed the harded value here

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1544/

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java [hunk 2]
```diff
@@ -536,7 +537,8 @@ public static Dictionary getDictionary(AbsoluteTableIdentifier absoluteTableIden
    */
   public static Map<String, List<Distributable>> nodeBlockMapping(List<Distributable> blockInfos) {
     // -1 if number of nodes has to be decided based on block location information
-    return nodeBlockMapping(blockInfos, -1);
+    return nodeBlockMapping(blockInfos, -1, null,
```
[the remark this answers is not in the document]
Comment (akashrn5) [reply processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java]: yes, in compaction case `BlockAssignmentStrategy.BLOCK_NUM_FIRST` is default , same as before

[the remark this answers is not in the document]
Comment (akashrn5) [reply processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java]: here i just did the refactoring

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1756/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1546/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1758/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonMergerRDD.scala [hunk 3]
```diff
@@ -401,7 +401,11 @@ class CarbonMergerRDD[K, V](
           .add(new CarbonInputSplitTaskInfo(entry._1, entry._2).asInstanceOf[Distributable])
     )
 
-    val nodeBlockMap = CarbonLoaderUtil.nodeBlockMapping(taskInfoList, -1)
+    // get all the active nodes of cluster and prepare the nodeBlockMap based on these nodes
+    val activeNodes = DistributionUtil
+      .ensureExecutorsAndGetNodeList(taskInfoList.asScala, sparkContext)
+
+    val nodeBlockMap = CarbonLoaderUtil.nodeBlockMapping(taskInfoList, -1, activeNodes.asJava)
```
[the remark this answers is not in the document]
Comment (akashrn5) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonMergerRDD.scala]: done

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1554/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9813/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1765/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1555/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9814/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1766/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**